packages/contracts/contracts/L1/rollup/CanonicalTransactionChain.sol

```diff
@@ -301,21 +303,14 @@ contract CanonicalTransactionChain is ICanonicalTransactionChain, Lib_AddressRes
             )
         );
 
-        bytes32 timestampAndBlockNumber;
-        assembly {
-            timestampAndBlockNumber := timestamp()
-            timestampAndBlockNumber := or(timestampAndBlockNumber, shl(40, number()))
-        }
-
-        IChainStorageContainer queueRef = queue();
-
-        queueRef.push(transactionHash);
-        queueRef.push(timestampAndBlockNumber);
-
-        // The underlying queue data structure stores 2 elements
-        // per insertion, so to get the real queue length we need
-        // to divide by 2 and subtract 1.
-        uint256 queueIndex = queueRef.length() / 2 - 1;
+        queueElements.push(
+            Lib_OVMCodec.QueueElement({
+                transactionHash: transactionHash,
+                timestamp: uint40(block.timestamp),
+                blockNumber: uint40(block.number)
```

Comment:
It should be safe to remove and save an extra SSTORE.

```
                timestamp: uint40(block.timestamp),
                blockNumber: uint40(block.number)
```

Probably worth doing this given we're already making significant changes to these data structures.

Reply:
OK, did that. 
It also resulted in a pretty big diff in `appendSequencerBatch` that you should take a look at. 
Basically, since we no longer have the `timestamp` and `number` available on the last queue element in the batch, instead of having an if/else to determine if we should use the values from the last queue element, we always use the current values from the EVM. I think this logic was actually just vestigial 'monotonicity code', that could have been removed anyways, but would appreciate you taking a look there.

Reply:
OK, did that.

It also resulted in a pretty big diff in appendSequencerBatch that you should take a look at.
Basically, since we no longer have the timestamp and number available on the last queue element in the batch, instead of having an if/else to determine if we should use the values from the last queue element, we always use the current values from the EVM. I think this logic was actually just vestigial 'monotonicity code', that could have been removed anyways, but would appreciate you taking a look there.